### PR TITLE
Fix serialization for IPC messages

### PR DESCRIPTION
**Pull request by andreibancioiu**
Fix serialization for IPC messages:
 - built-in function calls (vm output)
 - `getAllState`

### CODE REVIEW

File: arwen/contexts/output.go [hunk 1]
```diff
@@ -328,8 +328,8 @@ func (context *outputContext) AddToActiveState(rightOutput *vmcommon.VMOutput) {
 		rightOutput.GasRefund.Add(rightOutput.GasRefund, context.outputState.GasRefund)
 	}
 
-	for _, rightAccount := range rightOutput.OutputAccounts {
-		leftAccount, ok := context.outputState.OutputAccounts[string(rightAccount.Address)]
+	for address, rightAccount := range rightOutput.OutputAccounts {
```
Comment (sasurobert) [inline arwen/contexts/output.go:331]: please move it back :) it is much safer.

File: arwen/contexts/output.go [hunk 2]
```diff
@@ -345,11 +345,12 @@ func mergeVMOutputs(leftOutput *vmcommon.VMOutput, rightOutput *vmcommon.VMOutpu
 	if leftOutput.OutputAccounts == nil {
 		leftOutput.OutputAccounts = make(map[string]*vmcommon.OutputAccount)
 	}
-	for _, rightAccount := range rightOutput.OutputAccounts {
-		leftAccount, ok := leftOutput.OutputAccounts[string(rightAccount.Address)]
+
+	for address, rightAccount := range rightOutput.OutputAccounts {
```
Comment (sasurobert) [inline arwen/contexts/output.go:349]: please move it back :) it is much safer.

File: ipc/common/messagesBlockchain.go [hunk 3]
```diff
@@ -412,14 +412,14 @@ func NewMessageBlockchainProcessBuiltinFunctionRequest(callInput vmcommon.Contra
...
-	VMOutput *vmcommon.VMOutput
+	SerializableVMOutput *SerializableVMOutput
 }
 
 // NewMessageBlockchainProcessBuiltinFunctionResponse creates a response message
 func NewMessageBlockchainProcessBuiltinFunctionResponse(vmOutput *vmcommon.VMOutput, err error) *MessageBlockchainProcessBuiltinFunctionResponse {
 	message := &MessageBlockchainProcessBuiltinFunctionResponse{}
 	message.Kind = BlockchainProcessBuiltinFunctionResponse
-	message.VMOutput = vmOutput
+	message.SerializableVMOutput = NewSerializableVMOutput(vmOutput)
```
Comment (sasurobert) [inline ipc/common/messagesBlockchain.go:422]: add a unit test ?

File: arwen/contexts/output.go (hunk 1 above)
Comment (andreibancioiu) [reply arwen/contexts/output.go]: Reverted.

File: ipc/common/messagesBlockchain.go (hunk 3 above)
Comment (andreibancioiu) [reply ipc/common/messagesBlockchain.go:422]: Added.

File: arwen/contexts/output.go (hunk 2 above)
Comment (andreibancioiu) [reply arwen/contexts/output.go]: Reverted.

File: arwen/contexts/output.go (hunk 1 above)
Comment (camilbancioiu) [reply arwen/contexts/output.go]: Using `string(rightAccount.Address)` will still cause problems if there are duplicate `OutputAccounts`, due to random iteration over maps. It doesn't actually help, it only hides the problem. The duplication was the issue, not this iteration.

File: arwen/contexts/output.go (hunk 2 above)
Comment (camilbancioiu) [reply arwen/contexts/output.go]: See above.

**Pull request merged**